dropwizard-util/src/main/java/io/dropwizard/util/Size.java

```diff
@@ -3,39 +3,56 @@
 import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.Maps;
 
 import java.util.Locale;
 import java.util.Map;
+import java.util.TreeMap;
 import java.util.regex.Matcher;
 import java.util.regex.Pattern;
 
 import static com.google.common.base.Preconditions.checkArgument;
 import static java.util.Objects.requireNonNull;
 
+import java.util.Collections;
+
 public class Size implements Comparable<Size> {
     private static final Pattern SIZE_PATTERN = Pattern.compile("(\\d+)\\s*(\\S+)");
-
-    private static final Map<String, SizeUnit> SUFFIXES = new ImmutableMap.Builder<String, SizeUnit>()
-            .put("B", SizeUnit.BYTES)
-            .put("byte", SizeUnit.BYTES)
-            .put("bytes", SizeUnit.BYTES)
-            .put("KB", SizeUnit.KILOBYTES)
-            .put("KiB", SizeUnit.KILOBYTES)
-            .put("kilobyte", SizeUnit.KILOBYTES)
-            .put("kilobytes", SizeUnit.KILOBYTES)
-            .put("MB", SizeUnit.MEGABYTES)
-            .put("MiB", SizeUnit.MEGABYTES)
-            .put("megabyte", SizeUnit.MEGABYTES)
-            .put("megabytes", SizeUnit.MEGABYTES)
-            .put("GB", SizeUnit.GIGABYTES)
-            .put("GiB", SizeUnit.GIGABYTES)
-            .put("gigabyte", SizeUnit.GIGABYTES)
-            .put("gigabytes", SizeUnit.GIGABYTES)
-            .put("TB", SizeUnit.TERABYTES)
-            .put("TiB", SizeUnit.TERABYTES)
-            .put("terabyte", SizeUnit.TERABYTES)
-            .put("terabytes", SizeUnit.TERABYTES)
-            .build();
+    
+    private static final Map<String, SizeUnit> SUFFIXES;
+    
+    static {
+        final Map<String, SizeUnit> _suffixes = new ImmutableMap.Builder<String, SizeUnit>()
+                .put("B", SizeUnit.BYTES)
+                .put("byte", SizeUnit.BYTES)
+                .put("bytes", SizeUnit.BYTES)
+                .put("K", SizeUnit.KILOBYTES)
+                .put("KB", SizeUnit.KILOBYTES)
+                .put("KiB", SizeUnit.KILOBYTES)
+                .put("kilobyte", SizeUnit.KILOBYTES)
+                .put("kilobytes", SizeUnit.KILOBYTES)
+                .put("M", SizeUnit.MEGABYTES)
+                .put("MB", SizeUnit.MEGABYTES)
+                .put("MiB", SizeUnit.MEGABYTES)
+                .put("megabyte", SizeUnit.MEGABYTES)
+                .put("megabytes", SizeUnit.MEGABYTES)
+                .put("G", SizeUnit.GIGABYTES)
+                .put("GB", SizeUnit.GIGABYTES)
+                .put("GiB", SizeUnit.GIGABYTES)
+                .put("gigabyte", SizeUnit.GIGABYTES)
+                .put("gigabytes", SizeUnit.GIGABYTES)
+                .put("T", SizeUnit.TERABYTES)
+                .put("TB", SizeUnit.TERABYTES)
+                .put("TiB", SizeUnit.TERABYTES)
+                .put("terabyte", SizeUnit.TERABYTES)
+                .put("terabytes", SizeUnit.TERABYTES)
+                .build();
+        
+        final TreeMap<String, SizeUnit> treeSuffixes = Maps.newTreeMap(String.CASE_INSENSITIVE_ORDER);
```

Comment:
Why not simply use an [`ImmutableSortedMap`](http://docs.guava-libraries.googlecode.com/git/javadoc/com/google/common/collect/ImmutableSortedMap.html) in the first place?